pandas/plotting/_matplotlib/hist.py

```diff
@@ -144,14 +144,18 @@ def _make_plot(self) -> None:
                 kwds["label"] = self.columns
                 kwds.pop("color")
 
-            y = reformat_hist_y_given_by(y, self.by)
-
             # We allow weights to be a multi-dimensional array, e.g. a (10, 2) array,
             # and each sub-array (10,) will be called in each iteration. If users only
             # provide 1D array, we assume the same weights is used for all iterations
             weights = kwds.get("weights", None)
             if weights is not None and np.ndim(weights) != 1:
-                kwds["weights"] = weights[:, i]
+                weights = weights[:, i]
+
+            if weights is not None:
```

Comment:
Curious, why can't this block of code be in the the `if` statement above?

Reply:
it can, if the `if` above is split in two. Just trying to change as few lines as possible :)